src/wallet/rpcwallet.cpp

```diff
@@ -3952,21 +3952,18 @@ UniValue generate(const JSONRPCRequest& request)
             "\nMine up to nblocks blocks immediately (before the RPC call returns) to an address in the wallet.\n"
             "\nArguments:\n"
             "1. nblocks      (numeric, required) How many blocks are generated immediately.\n"
-            "2. private key  (hex string, required) private key in WIF to sign the generated blocks.\n"
+            "2. private key  (string, required) The aggregate private key to sign the generated blocks. The format is WIF.\n"
             "\nResult:\n"
             "[ blockhashes ]     (array) hashes of blocks generated\n"
             "\nExamples:\n"
             "\nGenerate 11 blocks signed with 1 privatekey\n"
-            + HelpExampleCli("generate", "11  \"c87509a1c067bbde78beb793e6fa76530b6382a4c0241e5e4a9ec0a0f44dc0d3\"")
+            + HelpExampleCli("generate", "11  \"L3wNcWW97SZYV6fb6b5VgoEvtybc9d1Uu6LeiQkVN4Z38VTivKT6\"")
         );
     }
 
     int num_generate = request.params[0].get_int();
 
-    std::vector<unsigned char> privkeyraw = ParseHex(request.params[1].get_str());
-    CKey cPrivKey;
-    cPrivKey.Set(privkeyraw.begin(), privkeyraw.end(), true);
-
+    CKey cPrivKey = DecodeSecret(request.params[1].get_str());
     if(!cPrivKey.size())
```

Comment:
It is more intuitive to use `cPrivKey.IsValid()` to check the validity of the private key. In addition, it should be checked if the private key corresponds to `FederationParams().GetLatestAggregatePubkey()`.

Reply:
fixed at 37cd8f627f64269901a0954934ecaa7e614a6407